src/ColorVectorSpace.jl

```diff
@@ -119,6 +96,47 @@ parametric(::Type{RGB24}, ::Type{N0f8}) = RGB24
 parametric(::Type{AGray32}, ::Type{N0f8}) = AGray32
 parametric(::Type{ARGB32}, ::Type{N0f8}) = ARGB32
 
+# Useful for leveraging iterator algorithms. Don't use this externally, as the implementation may change.
+channels(c::AbstractGray)    = (gray(c),)
+channels(c::TransparentGray) = (gray(c), alpha(c))
+channels(c::AbstractRGB)     = (red(c), green(c), blue(c))
+channels(c::TransparentRGB)  = (red(c), green(c), blue(c), alpha(c))
+
+nan(::Type{T}) where {T<:AbstractFloat} = convert(T, NaN)
+nan(::Type{C}) where {C<:MathTypes} = _nan(eltype(C), C)
+_nan(::Type{T}, ::Type{C}) where {T<:AbstractFloat,C<:AbstractGray} = (x = convert(T, NaN); C(x))
+_nan(::Type{T}, ::Type{C}) where {T<:AbstractFloat,C<:TransparentGray} = (x = convert(T, NaN); C(x,x))
+_nan(::Type{T}, ::Type{C}) where {T<:AbstractFloat,C<:AbstractRGB} = (x = convert(T, NaN); C(x,x,x))
+_nan(::Type{T}, ::Type{C}) where {T<:AbstractFloat,C<:TransparentRGB} = (x = convert(T, NaN); C(x,x,x,x))
+
+
+## Generic algorithms
+Base.add_sum(c1::MathTypes,c2::MathTypes) = mapc(Base.add_sum, c1, c2)
+Base.reduce_first(::typeof(Base.add_sum), c::MathTypes) = mapc(x->Base.reduce_first(Base.add_sum, x), c)
+function Base.reduce_empty(::typeof(Base.add_sum), ::Type{T}) where {T<:MathTypes}
+    z = Base.reduce_empty(Base.add_sum, eltype(T))
+    return zero(base_colorant_type(T){typeof(z)})
+end
+
+
+## Rounding & mod
+for f in (:trunc, :floor, :round, :ceil, :eps, :bswap)
+    @eval $f(g::Gray{T}) where {T} = Gray{T}($f(gray(g)))
+end
+eps(::Type{Gray{T}}) where {T} = Gray(eps(T))
+
+for f in (:trunc, :floor, :round, :ceil)
+    @eval $f(::Type{T}, g::Gray) where {T<:Integer} = Gray{T}($f(T, gray(g)))
```

Comment:
This seems to be an over-loose type annotation to me, since we don't really have `Gray{<:Integer}` types except for `Bool`.

https://github.com/JuliaGraphics/ColorTypes.jl/blob/0003cf95fc1ed348490d2a467ee95afea55ee9e0/src/types.jl#L344-L346

Making it 

```julia
$f(::Type{T}, g::Gray) where {T<:Integer} = $f(T, gray(g)))
```

is more reasonable if we consider that `T` specifies the output type for such usages.

cref: https://github.com/JuliaGraphics/ColorTypes.jl/issues/123#issuecomment-1272570990

cc: @myrddin89